rust/datafusion/src/physical_plan/repartition.rs

```diff
@@ -199,14 +211,12 @@ impl ExecutionPlan for RepartitionExec {
                     }
 
                     // notify each output partition that this input partition has no more data
-                    for channel in channels.iter_mut().take(num_output_partitions) {
-                        let tx = &mut channel.0;
+                    for (_, tx) in txs {
                         tx.send(None)
                             .map_err(|e| DataFusionError::Execution(e.to_string()))?;
                     }
                     Ok(())
                 });
-                tokio::task::yield_now().await;
```

Comment:
this is the workaround added in https://github.com/apache/arrow/pull/9580 and it is now removed in favor of what we think is the fix of the root cause